### ISSUE TITLE
Bridge libera.chat

### ISSUE THREAD

**Issue by TheEvilSkeleton** [matrix-org/matrix-appservice-irc#1324]
**Is your feature request related to a problem? Please describe.**

freenode has been acquired by a company, and the former owner switched to libera.chat.

**Describe the solution you'd like**

Bridge the new http://libera.chat/ network with appservice.

**Additional context**

https://www.kline.sh/

**Comment by ftyers** [comment]
It is underway. See [IRC Matrix Bridges](https://app.element.io/#/room/#irc:matrix.org).

**Comment by Half-Shot** [comment]
Hi folks, we're in talks with the libera folks but they are understandably massively busy starting up a brand new network.

**Comment by anarcat** [comment]
this is also being tracked in `#libera-matrix` on irc.libera.chat.

**Comment by eternaleye** [comment]
An update on Twitter: https://twitter.com/liberachat/status/1396920289641091079

**Comment by Half-Shot** [comment]
Yup, a lot of progress made on the call. A bit of work for me to do but it hopefully won't be the end of the world.

**Comment by AdamWill** [comment]
I'm sure everyone's working as hard as they can, but just to note, now freenode is shutting down channels that post links to libera, this has become a bit more urgent :( the freenode incarnations of several channels I follow have been zapped, so I guess I'll have to fire up an IRC client for a bit until the libera bridge is ready.

**Comment by Half-Shot** [comment]
Yup, I've personally dealt with a few Freenode takeovers myself today. We're hoping to have the bridge ready to go tomorrow.

**Comment by sanecito** [comment]
> We're hoping to have the bridge ready to go tomorrow.

Just so one can relevant resources lined up for tomorrow, is this something that will require a server update, and/or just an Element client update by whomever manages the room plumbing?

**Comment by Half-Shot** [comment]
@sanecito No update required on your side, it will be automatic.

**Comment by mcb30** [comment]
I'm seeing the dropdown offer `liberachat` as an option, and the list of channel operators is populated correctly, but adding the bridge is repeatably failing with `Bridge returned HTTP 500 Internal Server Error: Room failed validation. You may be attempting to "double bridge" this room. Error: 2`.  (I have already removed the Freenode bridge, so there are no existing IRC bridges).

Is this expected to work at this time, or is it still a work in progress?

Thanks!

**Comment by c-x-berger** [comment]
> adding the bridge is repeatably failing with `Bridge returned HTTP 500 Internal Server Error: Room failed validation. You may be attempting to "double bridge" this room. Error: 2`. (I have already removed the Freenode bridge, so there are no existing IRC bridges).

We're noticing something similar in `#purduelug:matrix.org`. Notably, traces of the Freenode bridge are apparently still part of the room state:

![image](https://user-images.githubusercontent.com/43154879/119909732-9cc74e00-bf23-11eb-99de-42972fc9fcf5.png)

**Comment by Half-Shot** [comment]
@mcb30 @c-x-berger can you report this in a new issue, we're trying to keep this one explictly about setting up the new bridge. Bugs / issues should go in separate issues.

**Comment by c-x-berger** [comment]
> can you report this in a new issue, we're trying to keep this one explictly about setting up the new bridge. Bugs / issues should go in separate issues.

Done. #1351

**Comment by TheEvilSkeleton** [comment]
Not exactly sure what is happening here. I read yesterday's blog post on Matrix and it mentioned that we can join libera.chat rooms: https://matrix.org/blog/2021/05/28/this-week-in-matrix-2021-05-28#liberachat-irc-network-news.

Does it mean the bridge is working?

**Comment by sanecito** [comment]
According to Libera.chat's fediverse account, it has been officially released:

https://fosstodon.org/@liberachat/106320476905973945

**Comment by TheEvilSkeleton** [comment]
I'll wait for a member of the team to close the issue.

**Comment by tohojo** [comment]
Hari Rana ***@***.***> writes:

> Not exactly sure what is happening here. I read yesterday's blog post on Matrix and it mentioned that we can join libera.chat rooms: https://matrix.org/blog/2021/05/28/this-week-in-matrix-2021-05-28#liberachat-irc-network-news.
>
> Does it mean the bridge is working?

Just tried it, and yeah that does indeed appear to be the case :)

It doesn't seem to work with SASL authentication though, so using
!storepass for automatic nickserv registration doesn't work :(

**Comment by alvinhochun** [comment]
> It doesn't seem to work with SASL authentication though, so using !storepass for automatic nickserv registration doesn't work :(

Just tried, `!storepass` worked fine for me.

**Comment by tohojo** [comment]
alvinhochun ***@***.***> writes:

>> It doesn't seem to work with SASL authentication though, so using !storepass for automatic nickserv registration doesn't work :(
>
> Just tried, `!storepass` worked fine for me.

Yeah, the command works, but does it auth you when you then subsequently
!reconnect ?

**Comment by tohojo** [comment]
Toke Høiland-Jørgensen ***@***.***> writes:

> alvinhochun ***@***.***> writes:
>
>>> It doesn't seem to work with SASL authentication though, so using !storepass for automatic nickserv registration doesn't work :(
>>
>> Just tried, `!storepass` worked fine for me.
>
> Yeah, the command works, but does it auth you when you then subsequently
> !reconnect ?

Ah, nvm, I forgot to also set !username; works now :)

**Comment by TheEvilSkeleton** [comment]
How did you people manage to create a libera.char account in Matrix (specifically Element)? Is it documented?

**Comment by Rspigler** [comment]
in #irc:matrix.org it says that the libera.chat bridge is still in testing?

**Comment by tohojo** [comment]
Hari Rana ***@***.***> writes:

> How did you people manage to create a libera.char account in Matrix
> (specifically Element)? Is it documented?

You don't need a libera.chat account; you just join the room from
whatever account you have already :)

**Comment by MightyCreak** [comment]
Do you know why #gamingonlinux doesn't appear in the room list?

**Comment by Tachi107** [comment]
@TheEvilSkeleton nope, I don't think it is documented, but I've actually followed you guide on Fedora Magazine, lol.
Instead of using `appservice-irc:matrix.org` I used `appservice:libera.chat`, and instead of `freenode_NickServ:matrix.org` I used `NickServ:libera.chat`.

**Comment by TheEvilSkeleton** [comment]
> > How did you people manage to create a libera.chat account in Matrix (specifically Element)? Is it documented?
> 
> @TheEvilSkeleton nope, I don't think it is documented, but I've actually followed you guide on Fedora Magazine, lol.
> Instead of using `appservice-irc:matrix.org` I used `appservice:libera.chat`, and instead of `freenode_NickServ:matrix.org` I used `NickServ:libera.chat`.

Yup, that's exactly what I did, and it worked for me.

**Comment by grahamperrin** [comment]
For reference: 

* https://matrix-org.github.io/matrix-appservice-irc/latest/usage.html#customising-your-experience

> … set a password to authenticate with services, …

– and part of a legacy page, through which I learnt what to do in response to **`err_needreggednick`**

* https://github.com/matrix-org/matrix-appservice-irc/wiki/End-user-FAQ#user-content-i-am-getting-kicked-out-of-a-bridged-room-with-the--err_needreggednick-error

**Comment by tohojo** [comment]
Creak ***@***.***> writes:

> Do you know why #gamingonlinux doesn't appear in the room list?

Rooms are created dynamically as users join them. So just tell your
Matrix client to join #gamingonlinux:libera.chat that it should
appear...

**Comment by cmurf** [comment]
From Element, if I start a chat with `@appservice-irc:matrix.org` and use `!storepass` I get a response that says "Successfully stored password for chat.freenode.net. You will now be reconnected to IRC." which is pretty confusing.

**Comment by grahamperrin** [comment]
> `@appservice-irc:matrix.org`

– is not for Libera Chat.

@ara4n hi, official https://matrix-org.github.io/matrix-appservice-irc/latest/usage.html#customising-your-experience is potentially confusing in that it makes an example **without** telling that the example was for Freenode. 

The beginning of deprecated https://github.com/matrix-org/matrix-appservice-irc/wiki/End-user-FAQ made a better example by naming Freenode.

**Comment by grahamperrin** [comment]
https://github.com/matrix-org/matrix-appservice-irc/issues/1324#issuecomment-850922098

> in `#irc:matrix.org` it says that the libera.chat bridge is still in testing?

True. 

Some confusion from <https://fosstodon.org/@liberachat/106320476905973945>, from which some people assumed that the bridge was officially released, ready for the masses: 

> … IRC gateways: our matrix bridge is now available! You can connect to …

The current topic at `ircs://irc.libera.chat/#libera-matrix` is: 

> Official Libera <> Matrix coordination, please be patient while the dust clears | Bridge is up and available for testing, be aware that it may be a bit wobbly while we work on issues | FREENODE BRIDGE UNLINK/DEOP ISSUES: email `support@matrix.org`, include room ID, problem description, and desired solution. Set a helpful subject line.

[Most recently](https://matrix.to/#/!vjOdckstTLwMkKHIgz:libera.chat/$vHzxwYTjmzZUQ92DRgu1DLzlaI-GYS2Bw1V9AZ60g4g?via=libera.chat&via=matrix.org&via=privacytools.io), in the channel: 

> The bridge with Libera isn't finished or officially released or anything, so issues are to be expected.

**Comment by Tachi107** [comment]
> From Element, if I start a chat with `@appservice-irc:matrix.org` and use `!storepass` I get a response that says "Successfully stored password for chat.freenode.net. You will now be reconnected to IRC." which is pretty confusing.

You have to chat with `@appservice:libera.chat`, see my comment: https://github.com/matrix-org/matrix-appservice-irc/issues/1324#issuecomment-851014890

**Comment by CRCinAU** [comment]
Is this completely busted at the moment?

I don't see anything coming across in #fedora or ##proxmox - the Matrix side is completely empty of content - but not empty of users.

**Comment by ljl-covid** [comment]
@CRCinAU while I'm not sure this is the cause of all these unidirectional failures, there is one caveat about the Matrix—Libera bridge that wasn't there with freenode: I understand due to privacy concerns, IRC messages are not relayed to Matrix unless and until all Matrix users in a room have successfully joined as their bridged IRC counterparts.

[This recent bug](https://github.com/matrix-org/matrix-appservice-irc/issues/1355) was caused by [the new behavior](https://github.com/matrix-org/matrix-appservice-irc/issues/1332) which traces back to [this 2016 bug report](https://github.com/matrix-org/matrix-appservice-irc/issues/212) and which I can see how it may end up losing a lot of messages during and after netsplits and such [as hinted here](https://github.com/matrix-org/matrix-appservice-irc/issues/212#issuecomment-484551054). I've already seen this behavior happening in #microg, but it's obviously too early to say how will this affect bridge stability in the long term. I suspect it was a strong request from Libera staff to the Matrix bridge people, to make sure no one on the Matrix side can *ever* see a message that their "IRC avatar" wasn't there to witness; however, in the future maybe individual rooms [could turn off this behavior](https://github.com/matrix-org/matrix-appservice-irc/issues/1332#issuecomment-847417685) if it's undesired because they are public, logged rooms.

In other ways, the Libera bridge is probably "cleaner" than the OFTC bridge, but this may be an important limitation to consider.

**Comment by tgr** [comment]
I get `Bridge returned HTTP 500 Internal Server Error: Room failed validation. You may be attempting to "double bridge" this room. Error: 2` when I try to bridge a Matrix room (`#wikipedia-hu-irc:matrix.org`) to a libera.chat channel (`#wikipedia-hu`). Not sure if that's an issue with the libera.chat bridge or something more generic.

**Comment by svenstaro** [comment]
Every now and then, the bridge appears to forget the channel keys I have for all of my channels that require them. This is quite annoying. Do I have to store them with the bridge somehow in a non-obvious way or is this currently simply broken?

**Comment by Mikaela** [comment]
@svenstaro as far as I am aware, the bridge won't store channel keys at all and you are hitting https://github.com/matrix-org/matrix-appservice-irc/issues/838 instead.

**Comment by LorbusChris** [comment]
Here's a small guide describing how to register with Libera.Chat and setup nick(s) through the Matrix bridge:
https://gist.github.com/LorbusChris/d44a5dd4671796cbf8528ca133dd909b

**Comment by TheEvilSkeleton** [comment]
There's also this one: https://kparal.wordpress.com/2021/06/01/connecting-to-libera-chat-through-matrix/

**Comment by daviddavis** [comment]
We have our rooms plumbed from matrix to libera. It's working really well but we are seeing some lag times of like 30+ seconds of matrix messages making it to libera.

**Comment by TheEvilSkeleton** [comment]
Am I the only one experiencing these?

![Screenshot_20210611-204435_Element](https://user-images.githubusercontent.com/50847364/121759905-f7e96b00-caf5-11eb-962a-968503eb746f.png)
![Screenshot_20210611-204443_Element](https://user-images.githubusercontent.com/50847364/121759906-f8820180-caf5-11eb-815c-06891c93acfc.png)

I always identify myself but this message keeps coming up.

**Comment by aaronraimist** [comment]
@TheEvilSkeleton did you give your password to the bridge with the `!storepass` command? If not then you have to manually identify yourself every time the bridge restarts.

**Comment by TheEvilSkeleton** [comment]
@aaronraimist yes I have. I forgot to mention that before.

**Comment by Tachi107** [comment]
> I always identify myself but this message keeps coming up.

I think it appears every time you reconnect (and maybe are safe to ignore). If you type INFO you should get your registered username

**Comment by chungy** [comment]
I also don't get automatically identified after using `!storepass`

**Comment by ThatGeoGuy** [comment]
> I also don't get automatically identified after using `!storepass`

I also had this problem but the solution is to make sure that you set your username since Libera won't auth via SASL without it. Your username can be set with `!username <your user name>` when chatting with the appservice. User names are usually the same as your nick, and all lowercase characters.

**Comment by Half-Shot** [comment]
Hi folks, the bridge is effectively live now and we're not going to monitor this thread for bug reports. We'd instead ask you interact via either the #libera-matrix:libera.chat Matrix room or create a new issue on this repo for us to triage.

**Comment by benks-io** [comment]
Did someone manage to connect a matrix room with a libera.chat room so that members can be connected with one of them, and all messages are synchronized?

**Comment by m1cl** [comment]
yes, it works as expected.

**Comment by MightyCreak** [comment]
> Did someone manage to connect a matrix room with a libera.chat room so that members can be connected with one of them, and all messages are synchronized?

I did, you can watch the GamingOnLinux room here: https://matrix.to/#/#mesamatrix-irc:matrix.org

**Comment by drybjed** [comment]
@MightyCreak Your explanation is for joining an existing IRC channel as a Matrix user, correct? Bridging is a bit different - both an IRC channel and Matrix room see the same messages and users. Any idea if that functionality works right now?

**Comment by benks-io** [comment]
@m1cl @MightyCreak Did you use [matrix-appservice-irc](https://matrix-org.github.io/matrix-appservice-irc) to bridge the rooms?

**Comment by MightyCreak** [comment]
Oops sorry, I've put the wrong room link (I've re-edited my post)

To answer your question, the room has been created by me and is bridged to an IRC chan

**Comment by alexxcons** [comment]
@TheEvilSkeleton  Automatic auth on reconnect to libera working here. Open a private chat with appservice:
```

!username <your NickServ username>
!storepass irc.libera.chat <your password>
```

More details here: https://meta.wikimedia.org/wiki/Matrix.org#Authenticating_with_LiberaChat

**Comment by benks-io** [comment]
@m1cl @MightyCreak Can you share your matrix-appservice-irc config file for bridging the room?

**Comment by MightyCreak** [comment]
@benks-io I don't have a config file, I'm merely using the IRC bridge proposed directly in Element

**Comment by benks-io** [comment]
I am looking for a way to synchronize the messages of two rooms on two servers with different protocols. Namely a Matrix room with matrix users on the one hand and an IRC Room on an IRC server on the other.
@MightyCreak So, your solution is an IRC room on a server with a matrix bridge, where Matrix Users may join but not the above?

**Comment by MightyCreak** [comment]
@benks-io  That's just it.

Create a room, clic on the "i" to get the information panel:

![image](https://user-images.githubusercontent.com/6382548/123103035-9e414480-d403-11eb-979c-80f36767c509.png)

Then clic on "Add widgets, etc..." And clic on "Add integrations" at the bottom-left:

![image](https://user-images.githubusercontent.com/6382548/123103188-c630a800-d403-11eb-9b5f-c426c04a01f0.png)

Select IRC bridge:

![image](https://user-images.githubusercontent.com/6382548/123103287-d6488780-d403-11eb-94bf-ef62d5b0204b.png)

And fill the form (someone on the other side need to have the rights to allow the matrix bot).

**Comment by benks-io** [comment]
@MightyCreak too easy! 🚀 Thanks for that. 🙏

**Comment by benks-io** [comment]
@MightyCreak That worked out until the step of confirmation. Wich IRC Client on libera-chat side did you use to accept the Invite?

**Comment by MightyCreak** [comment]
I wasn't on that side as I am not the moderator on the IRC chan, but I warned them before starting the process. As they didn't know much about Matrix, I guess it was easy to set up.

**Comment by legoktm** [comment]
It's been brought up in `#libera-matrix:libera.chat` that https://status.matrix.org/ has no entry for the Libera Chat bridge. Should it be listed there? Should I file a new ticket or support request for that?

**Comment by Xiretza** [comment]
> It's been brought up in `#libera-matrix:libera.chat` that https://status.matrix.org/ has no entry for the Libera Chat bridge. Should it be listed there?

I don't think so, because it's hosted and operated by libera, not matrix.org. On the other hand Gimpnet is also there, which runs on gnome.org, so I'm not sure. It should definitely be in https://github.com/matrix-org/matrix-appservice-irc/wiki/Bridged-IRC-networks though, so I opened #1415.

**Comment by Mikaela** [comment]
> I don't think so, because it's hosted and operated by libera, not matrix.org.

It is not hosted or operated by  Libera.Chat. It's hosted and operated by [Element Matrix Services](https://element.io/matrix-services) as can be seen e.g. by using [the federation tester](https://federationtester.matrix.org/api/report?server_name=libera.chat).

I don't know if EMS has a status page where it could be listed, but the bridge administration definitely has more to do with Matrix.org than Libera.Chat.

**Comment by heyakyra** [comment]
From my matrix.org account I was kicked from Libera.chat inactivity, but I'm unable to (re-)join any libera.chat rooms because of "No known servers" …any ideas?

**Comment by binarymaster** [comment]
@heyakyra this happened for me before, I was able to join again after cleaning up cache in Element.

**Comment by heyakyra** [comment]
Thanks! Is there a bug report though? Most users won't go through the trouble to dig this up

**Comment by tgr** [comment]
Given libera.chat was unbridged, should this task be reopened? Or is there a better place to look information?
